src/AutorestSwift/View Models/KeyValueViewModel.swift

```diff
@@ -32,15 +32,43 @@ import Foundation
 struct KeyValueViewModel {
     let key: String
     let value: String
+    // Flag indicates if value is optional
+    let optional: Bool
 
-    init(from param: Parameter, with model: CodeModel) {
-        self.key = param.serializedName!
-        if let constant = param.schema as? ConstantSchema {
-            self.value = "\"\(constant.value.value)\""
-        } else if let schema = model.schema(for: param.schema.name, withType: param.schema.type) {
-            self.value = schema.name
+    init(
+        from parameter: Parameter? = nil,
+        with codemodel: CodeModel? = nil,
+        and operation: Operation? = nil,
+        using key: String? = nil,
+        modelValue: String? = nil
+    ) {
```

Comment:
It seems like there's a lot of complexity in this initializer but I don't understand the reason for it. Perhaps we could break this into multiple initializers:

```swift 
init(from constant: ConstantSchema) {
let isString: Bool = constantSchema.valueType.type == AllSchemaTypes.string
  let val: String = constantSchema.value.value
  self.value = isString ? "\"\(val)\"" : "\(val)"
  self.optional = false
}

init(value: String, optional: Bool) {
  ...
}

init(from parameter: parameter, with model: CodeModel, and operation: Operation) {
  ...
}
```

Though I would still like to see a doc comment that explains where each initializer would be used.

Reply:
Bring down into 2 initializers.
1. Add logic to look up if the param is a Constant or a SignatureSignureParameter.  
if it is constant, the value will the constant value
If it is from Signature Parameter, the value will be name of the Signature Parameter.

2.Add a initializer just takes the key, value pair as Strings.
This is for building http header for contentType and Accept.  The values of them are taking from knownMediaType and MediaTypes of request and response.